runtime(doc): clarify smarttab (#17426)
vim/vim: runtime/doc/quickref.txt

'signcolumn'	  'scl'	    when to display the sign column
'smartcase'	  'scs'     no ignore case when pattern has uppercase
'smartindent'	  'si'	    smart autoindenting for C programs

> 'smarttab'	  'sta'     in indent, behave as if 'sts' = 'sw'

Reviewer: ```suggestion
'smarttab'	  'sta'     in indent insert 'shiftwidth' blanks for <Tab>
```

Author: With 'sta' the `<Tab>` key does not always insert 'sw' blanks, because `<Tab>` is always about going to the next tab stop, not about inserting X number of chars; 'sta' is also about the `<BS>` key, not only `<Tab>`.

The only thing I do not really like about the version I suggest, is that I had to shorten it too much keep it on a single line of 78 columns. Is that a strict rule in that file ?

Reviewer: > With 'sta' the <Tab> key does not always insert 'sw' blanks, because <Tab> is always about going to the next tab stop, not about inserting X number of chars; 'sta' is also about the <BS> key, not only <Tab>.

Yes true. 

> The only thing I do not really like about the version I suggest, is that I had to shorten it too much keep it on a single line of 78 columns. Is that a strict rule in that file ?

Yes it must fit into a 80 char (terminal) screen size. How about this then:

> 'smarttab'        'sta'     in indent insert blanks until next shiftwidth pos

Author: > 'smarttab' 'sta' in indent insert blanks until next shiftwidth pos

is even less clear to me. What is supposed to insert blanks ? What is a shiftwidth pos ?

There could be a kind of alternative as such
> 'smarttab' 'sta' in indent `<Tab>` and `<BS>` behave like `<C-t>` and `<C-d>`

The only problem is that it is false, `<C-t>` and `<C-d>` rewrite whitespace while `<Tab>` and `<BS>` really just go to the next tab stop.

I had o3 think for me and found (for a line of 79 chars):
> `<Tab>` in leading whitespace indents by 'shiftwidth'

which is close to the version you liked and is strictly better in my opinion: no suggestion of what should get inserted. And the verb ‘indent’ is vague enough to make the sentence not incorrect :-)
It also has the benefit of talking about what the user tries to achieve instead of what the option technically does.

I'll go ahead and rewrite all the files with that in mind.
